rbd/rbd_test.go

```diff
@@ -629,3 +629,43 @@ func TestTrashImage(t *testing.T) {
 	conn.DeletePool(poolname)
 	conn.Shutdown()
 }
+
+func TestImageMetadata(t *testing.T) {
+	conn, _ := rados.NewConn()
+	conn.ReadDefaultConfigFile()
+	conn.Connect()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	assert.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+
+	name := GetUUID()
+	image, err := rbd.Create(ioctx, name, 1<<22, 22)
+	assert.NoError(t, err)
+
+	err = image.Open()
+	assert.NoError(t, err)
+
+	metadataKey := "mykey"
+	metadataValue := "myvalue"
+
+	err = image.SetMetadata(metadataKey, metadataValue)
+	assert.NoError(t, err)
+	value, err := image.GetMetadata(metadataKey)
+	assert.NoError(t, err)
+	assert.Equal(t, metadataValue, value)
+	err = image.RemoveMetadata(metadataKey)
+	assert.NoError(t, err)
```

Comment:
Can you add a test to get the `metadatakey` again, and verify that the key is not set after removing it?

Reply:
I added the test